unit_tests/recovery/ut_database.cpp

```diff
@@ -34,11 +37,37 @@ ut_database_values_generator::ut_database_values_generator (const ut_database_co
                                ut_database_config::UPDATE_PAGE_DISCRETE_RATIO})
   // *INDENT-ON*
 {
+  // the log_lsa ctor actually does not set any value - and better not touch it
+  // values must be explicitely set
+  m_log_lsa.pageid = 10101; // arbitrary
+  m_log_lsa.offset = 0;
 }
 
-INT64 ut_database_values_generator::increment_and_get_lsa_log_id ()
+const log_lsa &ut_database_values_generator::increment_and_get_lsa_log ()
 {
-  return ++m_lsa_log_id;
+  const auto if_false_increment_page_id_else_increment_offset = m_rand_log_lsa_dist (m_gen);
+  if (if_false_increment_page_id_else_increment_offset == false)
+    {
+      // increment page id
+      assert (m_log_lsa.pageid < MAX_LOG_LSA_PAGEID);
+      ++m_log_lsa.pageid;
+      // just for fun, don't restart with 0-offset
```

Comment:
Fun is fun, but it may confuse next reader :). If the database behavior is any indication, the offset should be reset to 0 when advancing to next page.
```suggestion
      ++m_log_lsa.pageid;
      // just for fun, don't restart with 0-offset
```

Reply:
Corrected.